deploy/jbossas/modeshape-jbossas-subsystem/src/main/resources/schema/modeshape_3_0.xsd

```diff
@@ -574,7 +583,25 @@
             </xs:annotation>
         </xs:attribute>
     </xs:complexType>
-  
+
+    <xs:complexType name="file-persistence">
+        <xs:attribute name="path" type="xs:string" use="optional" default="${jboss.server.data.dir}/modeshape">
+            <xs:annotation>
+                <xs:documentation>
+                    The absolute path on the FS where the repository data should be persisted. If not specified,
+                    defaults to ${jboss.server.data.dir}/modeshape/${repository.name}
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
```

Comment:
As long as the "path" values can contain those environment variables, then using a single "path" is far better. Thanks.